### PR TITLE
Fix constrained optimization with `TPESampler`'s `constant_liar`

### PR DESCRIPTION
**Pull request by not522**
## Motivation
Using `constraints_enabled=True` and `constant_liar=True` may cause `scores` and `violations` to have different lengths.

## Description of the changes
- Fix `_get_observation_pairs`
- Add a test for correctly splitting trials

### CODE REVIEW

File: optuna/samplers/_tpe/sampler.py [hunk 1]
```diff
@@ -670,19 +670,22 @@ def _get_observation_pairs(
...
+                    violation = float("inf")
+                else:
+                    # Violation values of infeasible dimensions are summed up.
+                    violation = sum(v for v in constraint if v > 0)
+                violations.append(violation)
             else:
-                # Violation values of infeasible dimensions are summed up.
-                violation = sum(v for v in constraint if v > 0)
-            violations.append(violation)
+                violations.append(float("inf"))
```
Comment (not522) [inline optuna/samplers/_tpe/sampler.py:688]: `RUNNING` trials should be `above`.

File: tests/samplers_tests/tpe_tests/test_sampler.py [hunk 2]
```diff
@@ -894,6 +895,79 @@ def test_split_observation_pairs_with_constraints_below_include_infeasible() ->
...
+            system_attrs={_CONSTRAINTS_KEY: [-1]},
+        )
+    )
+
+    values, scores, violations = _tpe.sampler._get_observation_pairs(
+        study,
+        ["x"],
+        constant_liar,
+        constraints,
+    )
```
Comment (HideakiImamura) [inline tests/samplers_tests/tpe_tests/test_sampler.py:962]: How about asserting the length of `values`, `scores`, and `violations` are equal?

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**